cmd/memtierd/main.go

```diff
@@ -0,0 +1,397 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package main
+
+import (
+	"bufio"
+	"context"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+	"syscall"
+
+	"gopkg.in/yaml.v2"
+
+	"github.com/sirupsen/logrus"
+
+	"github.com/containerd/nri/pkg/api"
+	"github.com/containerd/nri/pkg/stub"
+)
+
+type config struct {
+	CfgParam1 string `json:"cfgParam1"`
+}
+
+type plugin struct {
+	stub stub.Stub
+	mask stub.EventMask
+}
+
+type MemtierdConfig struct {
+	Policy   Policy     `yaml:"policy"`
+	Routines []Routines `yaml:"routines"`
+}
+
+type Routines struct {
+	Name   string `yaml:"name"`
+	Config string `yaml:"config"`
+}
+
+type Policy struct {
+	Name   string `yaml:"name"`
+	Config string `yaml:"config"`
+}
+
+var (
+	cfg config
+	log *logrus.Logger
+)
+
+func (p *plugin) Configure(config, runtime, version string) (stub.EventMask, error) {
+	log.Infof("Connected to %s/%s...", runtime, version)
+
+	if config == "" {
+		return 0, nil
+	}
+
+	err := yaml.Unmarshal([]byte(config), &cfg)
+	if err != nil {
+		return 0, fmt.Errorf("failed to parse configuration: %w", err)
+	}
+
+	log.Info("Got configuration data %+v...", cfg)
```

Comment:
This should probably also be removed, if the plugin does not expect to get pre-launched by the runtime (by getting symlinked to /opt/nri/plugins/$idx-$plugin) and possibly get configuration data (stored in /etc/nri/conf.d).